### PR TITLE
[TC-355] Add CDN match checks when assigning or editing profiles

### PR DESCRIPTION
**Pull request by dg4prez**
Adds CDN mismatch/null checks to changing server profile in the UI and API.  Also checks for a mismatch against the servers with a particular profile assigned when changing that profile's CDN.

### CODE REVIEW

**Comment by dg4prez** [comment]
found an error, standby on this.

**Comment by mitchell852** [comment]
In the API, I see where you are checking this:

- when updating/creating a server, check that server.profile.cdn == server.cdn

but i don't see where you are checking this in the API:

- when updating a profile, check that profile.cdn == the cdn of the servers currently assigned to the profile

**Comment by dg4prez** [comment]
You're right Jeremy, I didn't add a check for that.  I'll put that in there as well.

**Comment by mitchell852** [comment]
Can you update the docs? https://github.com/apache/incubator-trafficcontrol/blob/master/docs/source/development/traffic_ops_api/v12/server.rst

specifically under POST /api/1.2/servers and PUT /api/1.2/servers/{:id}

just add a note next to the "profile" property description. something along the lines of "profile cdn must equal server cdn".

trying to improves these api docs a bit...

**Comment by dg4prez** [comment]
added and tested with sphinx make.

File: traffic_ops/app/lib/API/Server.pm [hunk 1]
```diff
@@ -911,6 +911,21 @@ sub is_server_valid {
...
+			$cdn_mismatch = 1;
+		} 
+		elsif ( $params->{cdnId} != $profile->cdn->id ) {
+			$cdn_mismatch = 1;
+		}
+	}
+
+	if ($cdn_mismatch) {
+		return ( 0, "CDN of profile does not match Server CDN" );
+	}
```
Comment (mitchell852) [inline traffic_ops/app/lib/API/Server.pm]: you could simplify all this code to simply:

if ($id) {
		my $profile = $self->db->resultset('Profile')->search( { 'me.id' => $params->{profileId}}, { prefetch => ['cdn'] } )->single();
		if ( !defined($profile->cdn) || $params->{cdnId} != $profile->cdn->id ) {
			return ( 0, "CDN of profile does not match Server CDN" );
		}
	}

**Pull request closed without merging**